wrap namespace name in card container

ShaiahWren opened the pull request:
Hi David!!
Issue:
https://issues.redhat.com/browse/AAH-165
I simply put flex-wrap on the .ns-card-container class. See cards.scss & namespace-card.tsx. 
![Screen Shot 2021-06-03 at 2 33 27 PM](https://user-images.githubusercontent.com/64337863/120695340-4a34e700-c479-11eb-9ae8-aed3678796d2.png)
![Screen Shot 2021-06-03 at 2 29 26 PM](https://user-images.githubusercontent.com/64337863/120695355-4ef99b00-c479-11eb-99bf-72147bfc6db9.png)

Seems to have done the trick.

ZitaNemeckova commented:
@ShaiahWren Not sure why it worked for you but it didn't for me in Chrome. 

This worked for me:
```diff
diff --git a/src/components/cards/cards.scss b/src/components/cards/cards.scss
index 65b03e3..fb356be 100644
--- a/src/components/cards/cards.scss
+++ b/src/components/cards/cards.scss
@@ -1,6 +1,6 @@
 .ns-card-container {
   width: 300px;
-  flex-wrap: wrap;
+  height: 280px;
 
   .title {
     font-weight: bold;
@@ -13,7 +13,10 @@
     width: 180px;
   }
 }
-
+.wrap {
+  width: 300px;
+  word-break: break-word;
+}
 .collection-card-container {
   width: 280px;
   // height: 340px;
diff --git a/src/components/cards/namespace-card.tsx b/src/components/cards/namespace-card.tsx
index b5914cc..66962cf 100644
--- a/src/components/cards/namespace-card.tsx
+++ b/src/components/cards/namespace-card.tsx
@@ -37,8 +37,8 @@ export class NamespaceCard extends React.Component<IProps, {}> {
             />
           </CardHeaderMain>
         </CardHeader>
-        <CardTitle>{company || name}</CardTitle>
-        <CardBody>{name}</CardBody>
+        <CardTitle className={"wrap"}>{company || name}</CardTitle>
+        <CardBody className={"wrap"}>{name}</CardBody>
         {namespaceURL && (
           <CardFooter>
             <Link to={namespaceURL}>View collections</Link>
```

ShaiahWren commented:
Hi Zita!!
I implemented your changes and here is what I have... it addresses the overflow for the title but not the CardBody of CardFooter. Is this a concern? 

David mentioned using a Tooltip + truncating method for the title. What do you think of that? 

See photo:
![Screen Shot 2021-06-04 at 10 24 03 AM](https://user-images.githubusercontent.com/64337863/120817148-730fb780-c51f-11eb-82e6-2609a8d671c8.png)

ShaiahWren commented:
And here is an edge case where a long long title completely overflows the card. Not a likely scenario, but the might have been what David was thinking when he thought of the Tooltip...
![Screen Shot 2021-06-04 at 10 33 08 AM](https://user-images.githubusercontent.com/64337863/120818282-75264600-c520-11eb-9551-6ccd28b110f3.png)

ZitaNemeckova commented:
@ShaiahWren You can create a namespace with name that contains spaces, commas and dots? It didn't let me make the name longer than 64 chars.
<img width="1108" alt="Screenshot 2021-06-04 at 16 39 05" src="https://user-images.githubusercontent.com/9210860/120819125-89d00180-c553-11eb-8241-d258bf00a86b.png">
<img width="1105" alt="Screenshot 2021-06-04 at 16 39 30" src="https://user-images.githubusercontent.com/9210860/120819131-8c325b80-c553-11eb-9353-1d870af324a3.png">

ZitaNemeckova commented:
hmm `CardFooter` should always be `View Collections` so it shouldn't be a problem. As for `CardBody` it should work with `clasName={'wrap'}`.

But truncating and having a Tooltip sounds like good idea too. Do you like that idea better?

ShaiahWren commented:
I think Tooltip sounds like a good option.
Shall I do that for both the title and CardBody, or just one?

ZitaNemeckova commented:
@ShaiahWren both :)

ShaiahWren commented:
I'm using the same truncating method on both the CardTitle and CardBody, but of course the "length" of the two will appear differently on bolded vs plain size text, see screenshot. Is this and issue? If so I'll make two separate truncating methods. Thanks! 
![Screen Shot 2021-06-07 at 10 59 51 AM](https://user-images.githubusercontent.com/64337863/121040099-a3559100-c77f-11eb-87ae-0292a5f174b7.png)

himdel commented:
So.. I don't want to block this PR on this, since we're already using this elsewhere.

**But** truncating the text to N characters will never really work... both of these are exactly 32 characters:

![20210607214932](https://user-images.githubusercontent.com/289743/121092154-a9c52680-c7da-11eb-8f5c-5077dd7ab6e4.png)
![20210607215130](https://user-images.githubusercontent.com/289743/121092156-ab8eea00-c7da-11eb-89d2-49bf7213e563.png)

Ideally, this should be fixed in css, using `text-overflow: ellipsis`

ZitaNemeckova commented on src/components/cards/namespace-card.tsx:
```diff
@@ -37,14 +39,30 @@ export class NamespaceCard extends React.Component<IProps, {}> {
         </CardHeader>
-        <CardTitle>{company || name}</CardTitle>
-        <CardBody>{name}</CardBody>
+        <Tooltip content={<CardTitle>{company || name}</CardTitle>}>
```
No need to add `<CardTitle>` or `<CardBody>` to the `content`.

ShaiahWren commented:
@ZitaNemeckova , I cleaned up the Tooltip so CardBody and CardTitle aren't in the 'content' attribute anymore. Also removed the className='wrap' from the footer. Not sure why I put that there in the first place, must have been from an earlier draft. 
@himdel you have a great point. In the case that we put text-overflow: ellipsis on CardTitle & CardBody, the full name will simply not be visible on the card view until clicking on 'view collections'. 
I'm happy to make the change. Lmk

ShaiahWren commented:
Pr closed
